### ISSUE TITLE
Rating: ARIA issues

### ISSUE THREAD

**Issue by miedrl** [primefaces/primefaces#8455]
**Describe the defect**
[Nu HTML Checker](https://validator.w3.org/nu/) can be used to validate a site's HTML syntax. The following validation errors appeared when using the rating component with the mentioned attributes. The errors seem to concern the tag <input type="hidden" .../> on the rating component.:

1. required="true" > 
![grafik](https://user-images.githubusercontent.com/77431692/154635335-c8bf7ae4-9b84-491d-9470-1179a74f1ade.png)

2. readonly="true" > 
![grafik](https://user-images.githubusercontent.com/77431692/154635403-856c4048-1425-4981-b492-89febcb43cb7.png)
![grafik](https://user-images.githubusercontent.com/77431692/154635424-622152ee-0b51-4d07-a789-a594c83e31be.png)

**Environment:**
 - PF Version: _10.0.0_

**To Reproduce**
1. Add the example code below to your application.
2. Copy your site's HTML code into the [Nu HTML Checker](https://validator.w3.org/nu/) and start validation.
3. See the validation errors.

**Expected behavior**
The attributes "aria-required", "aria-readonly" and "readonly" should not be added to the <input>-element within the rating component as they cause a HTML validation errors and thus violate [WCAG success criterion 4.1.1](https://www.w3.org/TR/WCAG21/#parsing) for accessibility.

**Example XHTML**
```html
<h:form id="frmTest">
    <p:rating value="3"
        required="true" />

     <p:rating value="3"
       readonly="true"/>
</h:form>
```

**Comment by melloware** [comment]
I know the issue. Will fix

**Comment by miedrl** [comment]
@melloware Thanks. Will the fix be available in version 12.0.0?

**Comment by melloware** [comment]
@miedrl I am seeing different results than you.  Can you run the NUHTML checker against the PF Showcase for Rating?

Use this URL: https://www.primefaces.org/showcase/ui/input/rating.xhtml

I am seeing...
```
Element [input](https://html.spec.whatwg.org/multipage/#the-input-element) is missing required attribute aria-valuenow.

Bad value slider for attribute role on element [input](https://html.spec.whatwg.org/multipage/#the-input-element).

 Attribute aria-disabled is unnecessary for elements that have attribute disabled.
```

**Comment by jepsar** [comment]
I do think we have an issue here though
https://github.com/primefaces/primefaces/blob/c98a9857c1c3c695ef6112532418e50fff9001f3/primefaces/src/main/java/org/primefaces/renderkit/InputRenderer.java#L76-L88

**Comment by melloware** [comment]
Yep I added that assuming it was harmless.  It really is harmless it just throws these strict checkers saying "you don't need this".

**Comment by melloware** [comment]
OK the readOnly and required issues were already fixed in 11.0 with this ticket: https://github.com/primefaces/primefaces/issues/7827

Which explains why I am not seeing them on 11.0.

**Comment by melloware** [comment]
Found this and it solves all ARIA issues with rating:
![image](https://user-images.githubusercontent.com/4399574/154772120-2bd31e59-9d5e-4abb-9540-aef849f2a6df.png)

**Comment by miedrl** [comment]
@melloware My answer might be too late now but when I run the NUHTML checker on the rating showcase, I get the same error. Fyi: You can run a bookmarklet with the following Javascript code to filter the messages from the NUHTML checker to only show WCAG 2.0 relevant errors. If run on the rating showcase there are no error messages left so it seems that the readonly error is fixed in 11.0.

`javascript:(function(){var removeNg=true;var filterStrings=["tag seen","Stray end tag","Bad start tag","violates nesting rules","Duplicate ID","first occurrence of ID","Unclosed element","not allowed as child of element","unclosed elements","not allowed on element","unquoted attribute value","Duplicate attribute"];var filterRE,root,results,result,resultText,i,cnt=0;filterRE=filterStrings.join("|");root=document.getElementById("results");if(!root){alert("No results container found.");return}results=root.getElementsByTagName("li");for(i=0;i<results.length;i++){result=results[i];if(result.className!==""){resultText=(result.innerText!==undefined?result.innerText:result.textContent)+"";if(resultText.match(filterRE)===null){result.style.display="none";result.className=result.className+" steveNoLike";cnt++}else if(removeNg==true){if(resultText.indexOf("not allowed on element")!==-1&&resultText.indexOf("ng-")!==-1){result.style.display="none";result.className=result.className+" steveNoLike";cnt++}}}}alert("Complete. "+cnt+" items removed.")})();`

**Comment by jepsar** [comment]
Might be the same issue as with https://github.com/primefaces/primefaces/pull/8462
